### PR TITLE
nixos/cloudflared: fix invalid systemd unit, required options with defaults

### PR DESCRIPTION
**Pull request by jyooru**
###### Description of changes

The module does not work without these changes.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bbigras — APPROVED** [review]
It evaluates, thanks!

File: nixos/modules/services/networking/cloudflared.nix [hunk 1]
```diff
@@ -263,7 +259,7 @@ in
       mapAttrs'
         (name: tunnel:
           nameValuePair "cloudflared-tunnel-${name}" ({
-            description = lib.mdDoc "Cloudflare tunnel '${name}' target";
+            description = "Cloudflare tunnel '${name}' target";
             requires = [ "cloudflared-tunnel-${name}.service" ];
             after = [ "cloudflared-tunnel-${name}.service" ];
             unitConfig.StopWhenUnneeded = true;
```
Comment (0x4A6F) [inline nixos/modules/services/networking/cloudflared.nix:265]: Useless parentheses around primitive expression:
```diff
     systemd.targets =
       mapAttrs'
         (name: tunnel:
-          nameValuePair "cloudflared-tunnel-${name}" ({
+          nameValuePair "cloudflared-tunnel-${name}" {
             description = "Cloudflare tunnel '${name}' target";
             requires = [ "cloudflared-tunnel-${name}.service" ];
             after = [ "cloudflared-tunnel-${name}.service" ];
             unitConfig.StopWhenUnneeded = true;
-          })
+          }
         )
         config.services.cloudflared.tunnels;
```

File: nixos/modules/services/networking/cloudflared.nix [hunk 2]
```diff
@@ -263,7 +259,7 @@ in
       mapAttrs'
         (name: tunnel:
           nameValuePair "cloudflared-tunnel-${name}" ({
-            description = lib.mdDoc "Cloudflare tunnel '${name}' target";
```
Comment (0x4A6F) [inline nixos/modules/services/networking/cloudflared.nix:266]: Good catch.

File: nixos/modules/services/networking/cloudflared.nix [hunk 3]
```diff
@@ -217,8 +214,7 @@ in
                 };
 
                 path = mkOption {
-                  type = with types; nullOr str;
-                  default = null;
```
Comment (0x4A6F) [inline nixos/modules/services/networking/cloudflared.nix:221]: With these changes I'm getting:
```
error: The option `services.cloudflared.tunnels.xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.ingress."hostname".path' is used but not defined.
```
Why are all these default values being removed?

Comment (jyooru) [reply nixos/modules/services/networking/cloudflared.nix:221]: Oops, I didn't realise this was for the submodule - in my configuration I'm just using `attrsOf str`. I've reverted this.

**Review by 0x4A6F — APPROVED** [review]

**Pull request merged**